ratelimit.go

```diff
@@ -35,24 +36,16 @@ import (
 // may block to throttle the goroutine.
 type Limiter interface {
 	// Take should block to make sure that the RPS is met.
-	Take() time.Time
-}
-
-// Clock is the minimum necessary interface to instantiate a rate limiter with
-// a clock or mock clock, compatible with clocks created using
-// github.com/andres-erbsen/clock.
-type Clock interface {
-	Now() time.Time
-	Sleep(time.Duration)
+	Take(context.Context) bool
```

Comment:
what does the `bool` argument mean?

Reply:
Right now, `Take` returns a `True` if it was unblocked without any cancellations from the passed in context, and `False` if it was unblocked because of the context.

Reply:
Can we update the documentation to make that part of the API

I'm also not sure how valuable it is, since the caller can just check `ctx.Err() != nil`

What is the value in having this in the API?